group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorShard.java

```diff
@@ -290,6 +294,61 @@ public GroupCoordinatorShard build() {
         }
     }
 
+    public static class DeleteShareGroupOffsetsResultHolder {
+        private final Errors topLevelError;
```

Comment:
I think you should separate Errors into `short topLevelErrorCode` and `String topLevelErrorMessage`. Otherwise, you'll be missing information about why an error occurs (not a share group, or non-existent group).